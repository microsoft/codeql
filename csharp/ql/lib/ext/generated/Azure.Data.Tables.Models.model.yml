# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
extensions:
  - addsTo:
      pack: codeql/csharp-all
      extensible: summaryModel
    data:
      - ["Azure.Data.Tables.Models", "TableAccessPolicy", False, "TableAccessPolicy", "(System.Nullable<System.DateTimeOffset>,System.Nullable<System.DateTimeOffset>,System.String)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Models.TableAccessPolicy.StartsOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableAccessPolicy", False, "TableAccessPolicy", "(System.Nullable<System.DateTimeOffset>,System.Nullable<System.DateTimeOffset>,System.String)", "", "Argument[1]", "Argument[this].Property[Azure.Data.Tables.Models.TableAccessPolicy.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableAccessPolicy", False, "TableAccessPolicy", "(System.Nullable<System.DateTimeOffset>,System.Nullable<System.DateTimeOffset>,System.String)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Models.TableAccessPolicy.Permission]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableAnalyticsLoggingSettings", False, "TableAnalyticsLoggingSettings", "(System.String,System.Boolean,System.Boolean,System.Boolean,Azure.Data.Tables.TableRetentionPolicy)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Models.TableAnalyticsLoggingSettings.Version]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableCorsRule", False, "TableCorsRule", "(System.String,System.String,System.String,System.String,System.Int32)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Models.TableCorsRule.AllowedOrigins]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableCorsRule", False, "TableCorsRule", "(System.String,System.String,System.String,System.String,System.Int32)", "", "Argument[1]", "Argument[this].Property[Azure.Data.Tables.Models.TableCorsRule.AllowedMethods]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableCorsRule", False, "TableCorsRule", "(System.String,System.String,System.String,System.String,System.Int32)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Models.TableCorsRule.AllowedHeaders]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableCorsRule", False, "TableCorsRule", "(System.String,System.String,System.String,System.String,System.Int32)", "", "Argument[3]", "Argument[this].Property[Azure.Data.Tables.Models.TableCorsRule.ExposedHeaders]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableErrorCode", False, "TableErrorCode", "(System.String)", "", "Argument[0]", "Argument[this].SyntheticField[Azure.Data.Tables.Models.TableErrorCode._value]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableErrorCode", False, "ToString", "()", "", "Argument[this].SyntheticField[Azure.Data.Tables.Models.TableErrorCode._value]", "ReturnValue", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableGeoReplicationStatus", False, "TableGeoReplicationStatus", "(System.String)", "", "Argument[0]", "Argument[this].SyntheticField[Azure.Data.Tables.Models.TableGeoReplicationStatus._value]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableGeoReplicationStatus", False, "ToString", "()", "", "Argument[this].SyntheticField[Azure.Data.Tables.Models.TableGeoReplicationStatus._value]", "ReturnValue", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableItem", False, "TableItem", "(System.String)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Models.TableItem.Name]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableModelFactory", False, "TableGeoReplicationInfo", "(Azure.Data.Tables.Models.TableGeoReplicationStatus,System.DateTimeOffset)", "", "Argument[1]", "ReturnValue.Property[Azure.Data.Tables.Models.TableGeoReplicationInfo.LastSyncedOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableModelFactory", False, "TableItem", "(System.String,System.String,System.String,System.String)", "", "Argument[0]", "ReturnValue.Property[Azure.Data.Tables.Models.TableItem.Name]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Models", "TableSignedIdentifier", False, "TableSignedIdentifier", "(System.String,Azure.Data.Tables.Models.TableAccessPolicy)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Models.TableSignedIdentifier.Id]", "value", "dfc-generated"]
