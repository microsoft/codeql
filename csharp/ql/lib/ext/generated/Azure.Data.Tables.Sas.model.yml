# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
extensions:
  - addsTo:
      pack: codeql/csharp-all
      extensible: summaryModel
    data:
      - ["Azure.Data.Tables.Sas", "TableAccountSasBuilder", False, "SetPermissions", "(System.String)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.Permissions]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasBuilder", False, "TableAccountSasBuilder", "(Azure.Data.Tables.Sas.TableAccountSasPermissions,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasBuilder", False, "TableAccountSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.Permissions]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasBuilder", False, "TableAccountSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "ToString", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_ExpiresOn", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_IPRange", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_Identifier", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_Permissions", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_Resource", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_ResourceTypes", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_Signature", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_StartsOn", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableAccountSasQueryParameters", False, "get_Version", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "SetPermissions", "(System.String)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.Permissions]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "TableSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableSasPermissions,System.DateTimeOffset)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.TableName]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "TableSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableSasPermissions,System.DateTimeOffset)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "TableSasBuilder", "(System.String,System.String,System.DateTimeOffset)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.TableName]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "TableSasBuilder", "(System.String,System.String,System.DateTimeOffset)", "", "Argument[1]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.Permissions]", "taint", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasBuilder", False, "TableSasBuilder", "(System.String,System.String,System.DateTimeOffset)", "", "Argument[2]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasIPRange", False, "TableSasIPRange", "(System.Net.IPAddress,System.Net.IPAddress)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasIPRange.Start]", "value", "dfc-generated"]
      - ["Azure.Data.Tables.Sas", "TableSasIPRange", False, "TableSasIPRange", "(System.Net.IPAddress,System.Net.IPAddress)", "", "Argument[1]", "Argument[this].Property[Azure.Data.Tables.Sas.TableSasIPRange.End]", "value", "dfc-generated"]
