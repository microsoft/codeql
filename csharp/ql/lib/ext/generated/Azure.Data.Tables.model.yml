# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
extensions:
  - addsTo:
      pack: codeql/csharp-all
      extensible: summaryModel
    data:
      - ["Azure.Data.Tables", "TableClient", False, "CreateQueryFilter", "(System.FormattableString)", "", "Argument[0].Property[System.FormattableString.Format]", "ReturnValue", "taint", "dfc-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "CreateQueryFilter", "(System.FormattableString)", "", "Argument[0]", "ReturnValue", "taint", "dfc-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "TableClient", "(System.String,System.String,Azure.Data.Tables.TableClientOptions)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "TableClient", "(System.Uri,System.String,Azure.Core.TokenCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "TableClient", "(System.Uri,System.String,Azure.Core.TokenCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[3]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "TableClient", "(System.Uri,System.String,Azure.Data.Tables.TableSharedKeyCredential)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", False, "TableClient", "(System.Uri,System.String,Azure.Data.Tables.TableSharedKeyCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "GetSasBuilder", "(Azure.Data.Tables.Sas.TableSasPermissions,System.DateTimeOffset)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "GetSasBuilder", "(System.String,System.DateTimeOffset)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "Query<T>", "(System.Linq.Expressions.Expression<System.Func<T,System.Boolean>>,System.Nullable<System.Int32>,System.Collections.Generic.IEnumerable<System.String>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "Query<T>", "(System.String,System.Nullable<System.Int32>,System.Collections.Generic.IEnumerable<System.String>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "QueryAsync<T>", "(System.Linq.Expressions.Expression<System.Func<T,System.Boolean>>,System.Nullable<System.Int32>,System.Collections.Generic.IEnumerable<System.String>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "QueryAsync<T>", "(System.String,System.Nullable<System.Int32>,System.Collections.Generic.IEnumerable<System.String>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableClient", True, "get_AccountName", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "CreateQueryFilter", "(System.FormattableString)", "", "Argument[0].Property[System.FormattableString.Format]", "ReturnValue", "taint", "dfc-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "CreateQueryFilter", "(System.FormattableString)", "", "Argument[0]", "ReturnValue", "taint", "dfc-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "TableServiceClient", "(System.String,Azure.Data.Tables.TableClientOptions)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "TableServiceClient", "(System.Uri,Azure.Core.TokenCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "TableServiceClient", "(System.Uri,Azure.Core.TokenCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[2]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "TableServiceClient", "(System.Uri,Azure.Data.Tables.TableSharedKeyCredential)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", False, "TableServiceClient", "(System.Uri,Azure.Data.Tables.TableSharedKeyCredential,Azure.Data.Tables.TableClientOptions)", "", "Argument[1]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "GetSasBuilder", "(Azure.Data.Tables.Sas.TableAccountSasPermissions,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[2]", "ReturnValue.Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "GetSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[0]", "ReturnValue.Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.Permissions]", "value", "dfc-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "GetSasBuilder", "(System.String,Azure.Data.Tables.Sas.TableAccountSasResourceTypes,System.DateTimeOffset)", "", "Argument[2]", "ReturnValue.Property[Azure.Data.Tables.Sas.TableAccountSasBuilder.ExpiresOn]", "value", "dfc-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "GetTableClient", "(System.String)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "Query", "(System.String,System.Nullable<System.Int32>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "QueryAsync", "(System.String,System.Nullable<System.Int32>,System.Threading.CancellationToken)", "", "Argument[1]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableServiceClient", True, "get_AccountName", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableSharedKeyCredential", False, "TableSharedKeyCredential", "(System.String,System.String)", "", "Argument[0]", "Argument[this].Property[Azure.Data.Tables.TableSharedKeyCredential.AccountName]", "value", "dfc-generated"]
      - ["Azure.Data.Tables", "TableUriBuilder", False, "TableUriBuilder", "(System.Uri)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Data.Tables", "TableUriBuilder", False, "ToUri", "()", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
