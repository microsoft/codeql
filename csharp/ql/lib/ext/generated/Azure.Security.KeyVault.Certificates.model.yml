# THIS FILE IS AN AUTO-GENERATED MODELS AS DATA FILE. DO NOT EDIT.
extensions:
  - addsTo:
      pack: codeql/csharp-all
      extensible: summaryModel
    data:
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", False, "CertificateClient", "(System.Uri,Azure.Core.TokenCredential,Azure.Security.KeyVault.Certificates.CertificateClientOptions)", "", "Argument[0]", "Argument[this].SyntheticField[Azure.Security.KeyVault.Certificates.CertificateClient._pipeline].SyntheticField[Azure.Security.KeyVault.KeyVaultPipeline.VaultUri]", "value", "dfc-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "GetCertificateOperation", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "GetCertificateOperationAsync", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartCreateCertificate", "(System.String,Azure.Security.KeyVault.Certificates.CertificatePolicy,System.Nullable<System.Boolean>,System.Collections.Generic.IDictionary<System.String,System.String>,System.Nullable<System.Boolean>,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartCreateCertificate", "(System.String,Azure.Security.KeyVault.Certificates.CertificatePolicy,System.Nullable<System.Boolean>,System.Collections.Generic.IDictionary<System.String,System.String>,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartCreateCertificateAsync", "(System.String,Azure.Security.KeyVault.Certificates.CertificatePolicy,System.Nullable<System.Boolean>,System.Collections.Generic.IDictionary<System.String,System.String>,System.Nullable<System.Boolean>,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartDeleteCertificate", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartDeleteCertificateAsync", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartRecoverDeletedCertificate", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "StartRecoverDeletedCertificateAsync", "(System.String,System.Threading.CancellationToken)", "", "Argument[this]", "ReturnValue", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificateClient", True, "get_VaultUri", "()", "", "Argument[this].SyntheticField[Azure.Security.KeyVault.Certificates.CertificateClient._pipeline].SyntheticField[Azure.Security.KeyVault.KeyVaultPipeline.VaultUri]", "ReturnValue", "value", "dfc-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificatePolicy", False, "CertificatePolicy", "(System.String,Azure.Security.KeyVault.Certificates.SubjectAlternativeNames)", "", "Argument[0]", "Argument[this]", "taint", "df-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificatePolicy", False, "CertificatePolicy", "(System.String,System.String)", "", "Argument[1]", "Argument[this].Property[Azure.Security.KeyVault.Certificates.CertificatePolicy.Subject]", "value", "dfc-generated"]
      - ["Azure.Security.KeyVault.Certificates", "CertificatePolicy", False, "CertificatePolicy", "(System.String,System.String,Azure.Security.KeyVault.Certificates.SubjectAlternativeNames)", "", "Argument[1]", "Argument[this].Property[Azure.Security.KeyVault.Certificates.CertificatePolicy.Subject]", "value", "dfc-generated"]
