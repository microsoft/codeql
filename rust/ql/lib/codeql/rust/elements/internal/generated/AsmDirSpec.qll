// generated by codegen, do not edit
/**
 * This module provides the generated definition of `AsmDirSpec`.
 * INTERNAL: Do not import directly.
 */

private import codeql.rust.elements.internal.generated.Synth
private import codeql.rust.elements.internal.generated.Raw
import codeql.rust.elements.internal.AstNodeImpl::Impl as AstNodeImpl

/**
 * INTERNAL: This module contains the fully generated definition of `AsmDirSpec` and should not
 * be referenced directly.
 */
module Generated {
  /**
   * An inline assembly direction specifier.
   *
   * For example:
   * ```rust
   * use core::arch::asm;
   * asm!("mov {input:x}, {input:x}", output = out(reg) x, input = in(reg) y);
   * //                                        ^^^                 ^^
   * ```
   * INTERNAL: Do not reference the `Generated::AsmDirSpec` class directly.
   * Use the subclass `AsmDirSpec`, where the following predicates are available.
   */
  class AsmDirSpec extends Synth::TAsmDirSpec, AstNodeImpl::AstNode {
    override string getAPrimaryQlClass() { result = "AsmDirSpec" }
  }
}
