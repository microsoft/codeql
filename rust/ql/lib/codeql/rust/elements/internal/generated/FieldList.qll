// generated by codegen, do not edit
/**
 * This module provides the generated definition of `FieldList`.
 * INTERNAL: Do not import directly.
 */

private import codeql.rust.elements.internal.generated.Synth
private import codeql.rust.elements.internal.generated.Raw
import codeql.rust.elements.internal.AstNodeImpl::Impl as AstNodeImpl

/**
 * INTERNAL: This module contains the fully generated definition of `FieldList` and should not
 * be referenced directly.
 */
module Generated {
  /**
   * A list of fields in a struct or enum variant.
   *
   * For example:
   * ```rust
   * struct S {x: i32, y: i32}
   * //       ^^^^^^^^^^^^^^^^
   * enum E {A(i32, i32)}
   * //     ^^^^^^^^^^^^^
   * ```
   * INTERNAL: Do not reference the `Generated::FieldList` class directly.
   * Use the subclass `FieldList`, where the following predicates are available.
   */
  class FieldList extends Synth::TFieldList, AstNodeImpl::AstNode { }
}
