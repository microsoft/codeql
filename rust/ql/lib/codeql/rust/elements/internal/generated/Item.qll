// generated by codegen, do not edit
/**
 * This module provides the generated definition of `Item`.
 * INTERNAL: Do not import directly.
 */

private import codeql.rust.elements.internal.generated.Synth
private import codeql.rust.elements.internal.generated.Raw
import codeql.rust.elements.internal.AddressableImpl::Impl as AddressableImpl
import codeql.rust.elements.MacroItems
import codeql.rust.elements.internal.StmtImpl::Impl as StmtImpl

/**
 * INTERNAL: This module contains the fully generated definition of `Item` and should not
 * be referenced directly.
 */
module Generated {
  /**
   * An item such as a function, struct, enum, etc.
   *
   * For example:
   * ```rust
   * fn foo() {}
   * struct S;
   * enum E {}
   * ```
   * INTERNAL: Do not reference the `Generated::Item` class directly.
   * Use the subclass `Item`, where the following predicates are available.
   */
  class Item extends Synth::TItem, StmtImpl::Stmt, AddressableImpl::Addressable {
    /**
     * Gets the attribute macro expansion of this item, if it exists.
     */
    MacroItems getAttributeMacroExpansion() {
      result =
        Synth::convertMacroItemsFromRaw(Synth::convertItemToRaw(this)
              .(Raw::Item)
              .getAttributeMacroExpansion())
    }

    /**
     * Holds if `getAttributeMacroExpansion()` exists.
     */
    final predicate hasAttributeMacroExpansion() { exists(this.getAttributeMacroExpansion()) }
  }
}
