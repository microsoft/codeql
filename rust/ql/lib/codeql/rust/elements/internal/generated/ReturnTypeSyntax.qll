// generated by codegen, do not edit
/**
 * This module provides the generated definition of `ReturnTypeSyntax`.
 * INTERNAL: Do not import directly.
 */

private import codeql.rust.elements.internal.generated.Synth
private import codeql.rust.elements.internal.generated.Raw
import codeql.rust.elements.internal.AstNodeImpl::Impl as AstNodeImpl

/**
 * INTERNAL: This module contains the fully generated definition of `ReturnTypeSyntax` and should not
 * be referenced directly.
 */
module Generated {
  /**
   * A return type notation `(..)` to reference or bound the type returned by a trait method
   *
   * For example:
   * ```rust
   * struct ReverseWidgets<F: Factory<widgets(..): DoubleEndedIterator>> {
   *     factory: F,
   * }
   *
   * impl<F> Factory for ReverseWidgets<F>
   * where
   *   F: Factory<widgets(..): DoubleEndedIterator>,
   * {
   *   fn widgets(&self) -> impl Iterator<Item = Widget> {
   *     self.factory.widgets().rev()
   *   }
   * }
   * ```
   * INTERNAL: Do not reference the `Generated::ReturnTypeSyntax` class directly.
   * Use the subclass `ReturnTypeSyntax`, where the following predicates are available.
   */
  class ReturnTypeSyntax extends Synth::TReturnTypeSyntax, AstNodeImpl::AstNode {
    override string getAPrimaryQlClass() { result = "ReturnTypeSyntax" }
  }
}
