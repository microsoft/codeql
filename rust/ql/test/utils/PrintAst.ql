/**
 * @name Print AST
 * @description Outputs a representation of a file's Abstract Syntax Tree
 * @id rust/test/print-ast
 * @kind graph
 */

import rust
import codeql.rust.printast.PrintAst
import codeql.rust.elements.internal.generated.ParentChild
import TestUtils

predicate shouldPrint(Locatable e) { toBeTested(e) }

import PrintAst<shouldPrint/1>
