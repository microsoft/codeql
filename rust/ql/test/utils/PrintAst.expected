lib.rs:
#    1| [SourceFile] SourceFile
